app/controllers/case_contacts/form_controller.rb

```diff
@@ -9,29 +9,21 @@ class CaseContacts::FormController < ApplicationController
 
   # wizard_path
   def show
-    authorize @case_contact
+    manage_form_step
     get_cases_and_contact_types
-    @page = wizard_steps.index(step) + 1
-    @total_pages = steps.count
-
     render_wizard
     wizard_path
   end
 
   def update
-    authorize @case_contact
+    manage_form_step(step_navigation: true)
     params[:case_contact][:status] = step.to_s if !@case_contact.active?
     remove_unwanted_contact_types
     remove_nil_draft_ids
-
     if CaseContactUpdateService.new(@case_contact).update_attrs(case_contact_params)
       respond_to do |format|
         format.html {
-          if step == steps.last
-            finish_editing
-          else
-            render_wizard @case_contact, {}, {case_contact_id: @case_contact.id}
-          end
+          manage_navigation
```

Comment:
New navigation behavior necessary for validating then routing to appropriate step

Reply:
I don't like this because it doesn't look like a typical controller action, I'd like to see the response behavior in this method, instead of finding out what's happens in manage_navigation...

What do you think of changing to smaller methods that are only responsible for setting variables, and leaving all the render/redirects logic in the action methods (show/update)?